### PR TITLE
feat(kpop): redesign [khcp-2007]

### PR DESCRIPTION
**Pull request by kaiarrowood**
### Summary
Update Kpop to match new designs

#### Changes made:
* Add `buttonText` prop - will generate a button if nothing passed in `default` slot
* Add `actions` and `footer` slot
* Update styles
* Update docs
* Update snaps

Figma: https://www.figma.com/file/sfSZhMfpzCzddEhd8tdk9s/Kong_Design-library?node-id=208%3A1336

image

### PR Checklist
- [x] Does not introduce dependencies
- [x] **Functional:** all changes do not break existing APIs and if so, bump major version.
- [x] **Tests pass:** check the output of yarn test packages/<Kongponent>
- [x] **Naming:** the files and the method and prop variables use the same naming conventions as other Kongponents
- [x] **Framework style:** abides by the essential rules in Vue's style guide
- [x] **Cleanliness:** does not have formatting issues, unused code (e.g., console.logs), or leftover comments
- [x] **Docs:** includes a technically accurate README, uses JSDOC where appropriate

### CODE REVIEW

File: docs/components/popover.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ For example a button:
 <KPop title="Cool header">
   <KButton>button</KButton>
   <template v-slot:content>
+  <KButton>text</KButton>
```
Comment (dustinryerson) [inline docs/components/popover.md:14]:
```suggestion
```

File: docs/components/popover.md [hunk 2]
```diff
@@ -19,13 +20,33 @@ For example a button:
 <KPop title="Cool header">
   <KButton>button</KButton>
   <template v-slot:content>
+
```
Comment (dustinryerson) [inline docs/components/popover.md:23]:
```suggestion
```

Comment (dustinryerson) [reply docs/components/popover.md:23]: We can remove this extra whitespace

File: docs/components/popover.md (hunk 1 above)
Comment (dustinryerson) [reply docs/components/popover.md:14]: There's an extra button in this example that wasn't there in the original:

**Original**
Screen Shot 2021-10-04 at 8 04 36 AM

**New**
Screen Shot 2021-10-04 at 8 02 56 AM

File: docs/components/popover.md [hunk 3]
```diff
@@ -420,6 +455,57 @@ This is the slot that takes in the content of the popover.
 </KPop>
 ```
 
+- `footer`
+This is an optional slot that takes in content for the footer bar.
+
+```vue
+<KPop title="Cool header">
+  <!-- Your element goes here -->
```
Comment (dustinryerson) [inline docs/components/popover.md:463]: This is pretty general, should we be more explicit about what element goes here?

File: packages/KLabel/KLabel.spec.js [hunk 4]
```diff
@@ -1,9 +1,20 @@
 import { mount } from '@vue/test-utils'
 import KLabel from '@/KLabel/KLabel'
 
+/**
+ * ALL TESTS MUST USE testMode: true
+ * We generate unique IDs for reference by aria properties. Debug mode strips these out
```
Comment (dustinryerson) [inline packages/KLabel/KLabel.spec.js:6]:
```suggestion
 * We generate unique IDs for reference by aria properties. Test mode strips these out
```

File: packages/KMultiselect/KMultiselect.spec.js [hunk 5]
```diff
@@ -3,11 +3,20 @@ import { default as KMultiselect } from './KMultiselect.vue'
 
 jest.useFakeTimers()
 
+/**
+ * ALL TESTS MUST USE testMode: true
+ * We generate unique IDs for reference by aria properties. Debug mode strips these out
```
Comment (dustinryerson) [inline packages/KMultiselect/KMultiselect.spec.js:8]:
```suggestion
 * We generate unique IDs for reference by aria properties. Test mode strips these out
```

File: packages/KPop/KPop.spec.js [hunk 6]
```diff
@@ -1,22 +1,61 @@
 import { mount, shallowMount } from '@vue/test-utils'
 import KPop from '@/KPop/KPop'
 
+/**
+ * ALL TESTS MUST USE testMode: true
+ * We generate unique IDs for reference by aria properties. Debug mode strips these out
```
Comment (dustinryerson) [inline packages/KPop/KPop.spec.js:6]:
```suggestion
 * We generate unique IDs for reference by aria properties. Test mode strips these out
```

File: packages/KoolTip/KoolTip.spec.js [hunk 7]
```diff
@@ -3,10 +3,19 @@ import KoolTip from '@/KoolTip/KoolTip'
 
 const positions = ['top', 'right', 'bottom', 'left']
 
+/**
+ * ALL TESTS MUST USE testMode: true
+ * We generate unique IDs for reference by aria properties. Debug mode strips these out
```
Comment (dustinryerson) [inline packages/KoolTip/KoolTip.spec.js:8]:
```suggestion
 * We generate unique IDs for reference by aria properties. Test mode strips these out
```

**Review by dustinryerson — CHANGES_REQUESTED** [review]
This is looking good! Left a few change suggestions and have a couple things to note:
1. The first interactive example seems to have an extra button in the KPop.
2. The `isSVG` example is broken. Originally, hovering the icons triggers the KPop, but now nothing is triggered on hover or click.
3. There is a missing spinner loading indicator in the last interactive example under `Events / Loading Content` section.

File: docs/components/popover.md (hunk 1 above)
Comment (kaiarrowood) [reply docs/components/popover.md]: Yep, this was left in from me testing keyboard nav. Removed now!

File: docs/components/popover.md (hunk 3 above)
Comment (kaiarrowood) [reply docs/components/popover.md:462]: Updated slot description text

File: docs/components/popover.md (hunk 2 above)
Comment (kaiarrowood) [reply docs/components/popover.md]: Done

File: packages/KLabel/KLabel.spec.js (hunk 4 above)
Comment (kaiarrowood) [reply packages/KLabel/KLabel.spec.js]: Got it

File: packages/KMultiselect/KMultiselect.spec.js (hunk 5 above)
Comment (kaiarrowood) [reply packages/KMultiselect/KMultiselect.spec.js]: Got it

File: packages/KPop/KPop.spec.js (hunk 6 above)
Comment (kaiarrowood) [reply packages/KPop/KPop.spec.js]: Got it

File: packages/KoolTip/KoolTip.spec.js (hunk 7 above)
Comment (kaiarrowood) [reply packages/KoolTip/KoolTip.spec.js]: Got it

**Review by dustinryerson — DISMISSED** [review]
@kaiarrowood all the requested changes look good except the missing spinner icon in the [Events / Loading Content](http://localhost:8080/components/popover.html#events-loading-content) section. The spinner icon is still missing on my local when I test it. Are you seeing the same?

**Review by dustinryerson — APPROVED** [review]
Rebuilding styles fixed the issue I was seeing with the spinner icon. LGTM!

**Pull request merged**